### PR TITLE
[WOR-486] Remove hardcoded Azure billing profile

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Context
We introduced a hardcoded Azure billing project in Dev and profile in April to enable testing of Azure workspaces while billing profile manager was being spun up. Now that BPM is up and we have real Azure billing projects, this hardcoded project is no longer necessary. 

There is a companion firecloud develop PR that cleans up the config values [here](https://github.com/broadinstitute/firecloud-develop/pull/3136)

## This PR
* Removes the codepaths that were injecting the hardcoded project, along with the associated config structures.
* Updates tests that were depending on the presence of the hardcoded project to instead create a "real" Azure billing project

---------
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aherbst-broad** [comment]
jenkins retest

File: core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala [hunk 1]
```diff
@@ -218,39 +185,6 @@ class BillingProfileManagerDAOSpec extends AnyFlatSpec with MockitoSugar {
     assertResult(Seq(expectedApp))(apps)
   }
 
-  it should "return all profiles from listBillingProfiles when the profiles exceeds the request batch size" in {
```
Comment (cahrens) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala]: I don't think we want to lose this test case.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala [hunk 2]
```diff
@@ -56,9 +51,9 @@ class BillingProfileManagerDAOSpec extends AnyFlatSpec with MockitoSugar {
   )
   val testContext = RawlsRequestContext(userInfo)
 
-  behavior of "listBillingProfiles"
+  behavior of "getAllBillingProfiles"
 
-  it should "return billing profiles to which the user has access" in {
+  it should "return all profiles from listBillingProfiles when the profiles exceeds the request batch size" in {
```
Comment (aherbst-broad) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala:56]: This diff is a bit confusing, but the this is the result of deleting the `return billing profiles to which the user has access` test; I believe it was really only exercising that we were getting the hardcoded billing profile back and the `return all profiles from listBillingProfiles when the profiles exceeds the request batch size` test exercises the listing functionality sufficiently.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionSpec.scala [hunk 3]
```diff
@@ -7,9 +7,19 @@ import akka.testkit.TestKit
 import com.google.api.client.auth.oauth2.Credential
 import org.broadinstitute.dsde.rawls.config.MethodRepoConfig
 import org.broadinstitute.dsde.rawls.dataaccess.{drs, _}
-import org.broadinstitute.dsde.rawls.dataaccess.drs.{DrsHubMinimalResponse, DrsHubResolver, ServiceAccountEmail, ServiceAccountPayload}
```
Comment (aherbst-broad) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionSpec.scala:10]: This is all scalafmt, not sure how it got missed by the global reformat last month.

File: model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 4]
```diff
@@ -246,13 +246,13 @@ object Workspace {
...
   }
 
-  def apply(namespace: String,
-            name: String,
-            workspaceId: String,
-            createdDate: DateTime,
-            lastModified: DateTime,
-            createdBy: String,
-            attributes: AttributeMap
+  def buildMcWorkspace(namespace: String,
```
Comment (aherbst-broad) [inline model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala:249]: I was getting turned around why workspace objects being instantiated via the apply() method were getting a type of `mc` in the DB; this hopefully makes it more explicit and less prone to misuse.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala (hunk 1 above)
Comment (aherbst-broad) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala]: Thanks; this was an intermediate diff, the test is back.

**Comment by aherbst-broad** [comment]
jenkins retest

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -39,7 +39,7 @@ jobs:
 
       - name: Secrets check
         run: |
-          ./minnie-kenny.sh --force
+          bash -x ./minnie-kenny.sh --force
```
Comment (cahrens) [inline .github/workflows/test.yml]: Was this just experimentation?

File: core/src/test/resources/reference.conf [hunk 6]
```diff
@@ -112,10 +112,6 @@ resourceBuffer {
 multiCloudWorkspaces {
     enabled = true
     azureConfig {
-        spendProfileId = "wm-spend-profile-id"
-        subscriptionId =  "50d04424-1ff9-11ed-861d-0242ac120002"
-        resourceGroupId = "fake_mrg"
-        tenantId =  "50d04424-1ff9-11ed-861d-0242ac120003"
         alphaFeatureGroup = "fake_alpha_group"
         billingProjectName = "fake_mc_billing_project_name"
```
Comment (cahrens) [inline core/src/test/resources/reference.conf]: delete `billingProjectName` also, correct?

File: core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala [hunk 7]
```diff
@@ -121,6 +120,24 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Test
...
+    val billingProject = RawlsBillingProject(RawlsBillingProjectName("test-azure-bp"),
+                                             CreationStatuses.Ready,
+                                             None,
+                                             None,
+                                             billingProfileId = Some(UUID.randomUUID().toString)
+    )
+    when(workspaceService.withBillingProjectContext(any(), any())(any())).thenAnswer { invocation =>
+      (invocation.getArgument(2): RawlsBillingProject => Future[Workspace])(billingProject)
+    }
+    when(bpDAO.getBillingProfile(any[UUID], any[RawlsRequestContext])).thenReturn(
```
Comment (cahrens) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala:132]: Are these changes necessary because before the hardcoded project was being returned?

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala [hunk 8]
```diff
@@ -1110,64 +1123,34 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
...
+          workspaceId = wsId.toString,
+          createdDate = DateTime.now,
+          lastModified = DateTime.now,
+          createdBy = "testuser@example.com",
+          attributes = Map()
         )
-          .thenReturn(
-            new CreateCloudContextResult().jobReport(new JobReport().id("fake_id").status(StatusEnum.SUCCEEDED))
-          )
-        when(services.workspaceManagerDAO.createAzureRelay(any[UUID], any[String], any[RawlsRequestContext]))
```
Comment (cahrens) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala:1131]: Why is all of this unnecessary now?

File: .github/workflows/test.yml (hunk 5 above)
Comment (cahrens) [reply .github/workflows/test.yml]: Note that [this change](https://github.com/broadinstitute/rawls/pull/2083/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R42) may help… it's hard to know for sure yet.

Comment (cahrens) [reply .github/workflows/test.yml]: Ugh, that commit failed on merge… back to the drawing board.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala (hunk 7 above)
Comment (aherbst-broad) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala:132]: Correct.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala (hunk 8 above)
Comment (aherbst-broad) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala:1131]: I felt putting the workspace in the DB directly rather than mocking the needed dependencies and then calling the POST lent itself to a simpler test.

**Review by cahrens — APPROVED** [review]

**Review by blakery — APPROVED** [review]
